shared/macros/10-oval.jinja

```diff
@@ -1180,3 +1180,141 @@ Generates the :code:`<affected>` tag for OVAL check using correct product platfo
+        <criterion test_ref="test_{{{ rule_id }}}_password_max_life_not_empty"
+          comment="Passwords must have the maximum password age set non-empty in /etc/shadow."/>
+    </criteria>
+  </definition>
+
+  <!-- Define a test for the /etc/shadow file for non-system accounts to look for the maximum
+       password change interval. -->
+  {{{ textfilecontent54_test_etc_shadow(
+    test_id="test_" + rule_id + "_password_max_life_existing",
+    regex=regex,
+    external_variable_id="var_accounts_maximum_age_login_defs",
+    operation="less than or equal") }}}
+
+  <!-- Define a second test to ensure the maximum password life is at least the defined
+       minimum (usually 1). -->
+  {{{ textfilecontent54_test_etc_shadow(
+    test_id="test_" + rule_id + "_password_max_life_existing_minimum",
+    regex=regex,
+    external_variable_id="var_accounts_minimum_age_login_defs",
+    operation="greater than or equal") }}}
+
+  <ind:textfilecontent54_test id="test_{{{ rule_id }}}_password_max_life_not_empty" version="1"
+    check="all" check_existence="none_exist"
+    comment="Passwords must have the maximum password age set non-empty in /etc/shadow.">
+    <ind:object object_ref="object_{{{ rule_id }}}_shadow_password_users_max_life_not_existing"/>
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_{{{ rule_id }}}_shadow_password_users_max_life_not_existing" version="1">
+    <ind:filepath>/etc/shadow</ind:filepath>
+    <!-- filters away accounts with no passwords and locked passwords (passwords are located in the 2nd field)-->
+    <ind:pattern operation="pattern match">{{{ regex_empty }}}</ind:pattern>
+    <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+</def-group>
+{{%- endmacro %}}
+
+{{#
+Generates an OVAL test used in checks a particular field in the "/etc/shadow" file.
+
+:param test_id: OVAL test ID
+:type test_id: str
+:param regex: Regex that matches the entries in shadow file and captures the particular field value to a subexpression
+:type regex: str
+:param external_variable_id: External variable ID
+:type external_variable_id: str
+:param operation: operation
+:type operation: str
+
+#}}
+{{%- macro textfilecontent54_test_etc_shadow(test_id, regex, external_variable_id, operation) -%}}
+
+  <ind:textfilecontent54_test id="{{{ test_id }}}" version="1"
+    check="all" check_existence="any_exist"
+    comment="Compares a specific field in /etc/shadow with a specific variable value">
+    <ind:object object_ref="object_{{{ test_id }}}"/>
+    <ind:state state_ref="state_{{{ test_id }}}"/>
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_{{{ test_id }}}" version="1">
+    <ind:filepath>/etc/shadow</ind:filepath>
+    <!-- filters away accounts with no passwords and locked passwords (passwords are located in the 2nd field)-->
```

Comment:
@jan-cerny  I don't understand this comment. Can you explain please? It confuses me because the next regexp is supplied in form of macro argument.

Reply:
You're right that it's confusing. It's left there from the previous version where this wasn't a macro. Although in the current state it generates a correct statement in the generated OVAL, it doesn't make sense to have it here. It would be better to remove it and instead add a similar comment to the place where this is called.